app/models/role.rb

```diff
@@ -129,6 +137,15 @@ def archived?
     archived_at.present?
   end
 
+  def nextcloud_group
+    FeatureGate.assert! 'groups.nextcloud'
+
+    case (setting = self.class.nextcloud_group)
+    when String then { 'gid' => "hitobito-#{setting}", 'displayName' => setting }
```

Comment:
As far as I understand it `gid` is the group id. I could also only pass the names and the nextcloud-plugin that does the mapping generates a sha256 from the name to have a unique ID. Since the group-names can change, I prefer this.

Reply:
Do you intend to serve two different use cases with the boolean vs. string? If I specify a string (e.g. "Trainers" in SKV), and there are more than one group of that type in hitobito, all people from all these groups will go to the same nextcloud group. On the other hand, if I specify just `true`, all people from these different trainer groups will be put into separate nextcloud groups. Seems confusing to me as a developer: usually `true` is just a shorthand for the default behaviour in such DSLs, but here there is no way to acheive with a string the same behaviour as `true` produces.
If both ways are definitely needed, how about letting the role class define a method or Proc class attr instead, which can generate the nextcloud group id and display name based on the role instance? This would also be more extensible for the future, if we e.g. ever need to put the user into multiple nextcloud groups based on just one hitobito role.

Reply:
Currently, there are the two use-cases you identified. For the SKV we have several Fachkommissionen, which should map to different nextcloud-groups.

The current method could be extended to support a Symbol (pointing to a method) or a Proc (which could be called). I will try if this is as easy as I think it is.

Reply:
It was as easy, though I am unsure what "can of worms" I opened with this flexibility. It's not a pandoras box, but can break the integration in a very subtle and hard-to-debug way. I tried to document the class_attribute better, in order to at least provide some help.